### PR TITLE
Set changesetId in BriefcaseConnection on pullAndMergeChanges

### PR DESCRIPTION
**Pull request by kabentley**
The member `changeSetId` of `BriefcaseConnection` should be updated after `pullAndMergeChanges`.

There are currently no tests for BriefcaseConnection that use IModelHub. We will have to create that framework, but I'll do it in another PR.

### CODE REVIEW

File: core/frontend/src/BriefcaseConnection.ts [hunk 1]
```diff
@@ -232,7 +232,7 @@ export class BriefcaseConnection extends IModelConnection {
    */
   public async pullAndMergeChanges(version?: IModelVersionProps): Promise<void> {
     this.requireTimeline();
-    return IpcApp.callIpcHost("pullAndMergeChanges", this.key, version);
+    this._changeSetId = await IpcApp.callIpcHost("pullAndMergeChanges", this.key, version);
```
Comment (pmconne) [inline core/frontend/src/BriefcaseConnection.ts:235]: Do you think applications might want to be notified via an event when the changesetId changes?

Comment (pmconne) [reply core/frontend/src/BriefcaseConnection.ts:235]: `BriefcaseTxns.onChangesPulled` exists but I'm not certain the ordering between that event being dispatched and the IModelConnection's changesetId being updated.

File: core/backend/src/BriefcaseManager.ts [hunk 2]
```diff
@@ -87,7 +85,7 @@ export class BriefcaseManager {
 
   /** @internal */
   public static getChangeSetFolderNameFromId(changeSetId: string): string {
-    return changeSetId || this._firstChangeSetDir;
+    return changeSetId || "first";
```
Comment (pmconne) [inline core/backend/src/BriefcaseManager.ts:88]: What is this function for? I found no callers in imodeljs repo.

Comment (kabentley) [reply core/backend/src/BriefcaseManager.ts:88]: I don't know what it was there for, I'll remove it.

File: core/frontend/src/BriefcaseConnection.ts (hunk 1 above)
Comment (kabentley) [reply core/frontend/src/BriefcaseConnection.ts:235]: the `onChangesPulled` event will happen before the `BriefcaseConnection`'s member is updated.

File: core/backend/src/BriefcaseManager.ts (hunk 2 above)
Comment (kabentley) [reply core/backend/src/BriefcaseManager.ts]: gone

**Review by pmconne — APPROVED** [review]

**Comment by scsewall** [comment]
Perhaps `BriefcaseDb.pullAndMergeChanges` should get the same improvement?

**Comment by kabentley** [comment]
> Perhaps `BriefcaseDb.pullAndMergeChanges` should get the same improvement?

It [already does](https://github.com/imodeljs/imodeljs/blob/846a0fb85b2d25f559416082b1ac9c7d05f52423/core/backend/src/IModelDb.ts#L2325)

**Comment by scsewall** [comment]
> > Perhaps `BriefcaseDb.pullAndMergeChanges` should get the same improvement?
> 
> It [already does](https://github.com/imodeljs/imodeljs/blob/846a0fb85b2d25f559416082b1ac9c7d05f52423/core/backend/src/IModelDb.ts#L2325) set the changesetId. But i suppose it could return it too. Is that what you mean?

Yes, I was talking about the return value.

**Pull request merged**